### PR TITLE
SAK-40325 - Error changing "Grade Scale" of an assignment

### PR DESCRIPTION
**Pull request by adrianmticarum**

### CODE REVIEW

**Comment by jonespm** [comment]
@adrianmticarum Hi! In the future could you add the title for what's being fixed to the commits? Thanks!

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 1]
```diff
@@ -7044,31 +7044,29 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
...
-                                Assignment assignment = assignmentService.getAssignment(assignmentId);
-                                if (assignment != null) {
-                                    scaleFactor = assignment.getScaleFactor();
-                                }
-                            }
-                        } catch (IdUnusedException | PermissionException e) {
-                            log.error(e.getMessage());
-                        }
+			if (gradePoints != null && gradeType == SCORE_GRADE_TYPE) {
+				if ((gradePoints.length() == 0)) {
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: can remove double paren

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -7044,31 +7044,29 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
...
-                        state.setAttribute(NEW_ASSIGNMENT_GRADE_POINTS, gradePoints);
-                    }
-                }
-            }
+					validPointGrade(state, gradePoints, scaleFactor);
+					// when scale is points, grade must be integer and less than maximum value
+					gradePoints = scalePointGrade(state, gradePoints, scaleFactor);
+					state.setAttribute(NEW_ASSIGNMENT_GRADE_POINTS, gradePoints);
+				}
+			}
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: spacing issue

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 3]
```diff
@@ -7044,31 +7044,29 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
...
+			if (gradePoints != null && gradeType == SCORE_GRADE_TYPE) {
+				if ((gradePoints.length() == 0)) {
+					// in case of point grade assignment, user must specify maximum grade point
+					addAlert(state, rb.getString("plespethe3"));
+				} else {
+					Integer scaleFactor = assignmentService.getScaleFactor();
+					try {
+						if (StringUtils.isNotEmpty(assignmentRef)) {
+							Assignment assignment = assignmentService.getAssignment(assignmentId);
+							if (assignment != null && assignment.getScaleFactor() != null) {
```
Comment (ern) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: if the scale factor on the assignment is null should use:
org.sakaiproject.assignment.impl.AssignmentServiceImpl#getScaleFactor

Comment (adrianmticarum) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Before the try block (line 7052) we are already getting the scaleFactor from assignmentService.

Comment (ern) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: ok looks good

**Pull request merged**